src/System.Runtime.Analyzers/UnitTests/SpecifyCultureInfoTests.cs

```diff
@@ -9,12 +10,480 @@ public class SpecifyCultureInfoTests : DiagnosticAnalyzerTestBase
+        }
+
+        [Fact]
+        public void CA1304_MethodOverloadCount3_CSharp()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Globalization;
+
+public class CultureInfoTestClass2
+{
+    public static void Method()
+    {
+        MethodOverloadCount3();
+    }
+
+    public static void MethodOverloadCount3()
+    {
+        MethodOverloadCount3(CultureInfo.CurrentCulture);
+    }
+
+    public static void MethodOverloadCount3(CultureInfo provider)
+    {
+        Console.WriteLine(string.Format(provider, """"));
+    }
+    public static void MethodOverloadCount3(string b)
+    {
+    }
+}",
+            GetCSharpResultAt(9, 9, SpecifyCultureInfoAnalyzer.Rule, "CultureInfoTestClass2.MethodOverloadCount3()", "CultureInfoTestClass2.Method()", "CultureInfoTestClass2.MethodOverloadCount3(System.Globalization.CultureInfo)"));
+        }
+
+        [Fact]
+        public void CA1304_MethodOverloadWithJustCulltureInfoAsExtraParameter_CSharp()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Globalization;
+
+public class CultureInfoTestClass2
+{
+    public static void Method()
+    {
+        MethodOverloadWithJustCulltureInfoAsExtraParameter(2, 3);
+    }
+
+    public static void MethodOverloadWithJustCulltureInfoAsExtraParameter(int a, int b)
+    {
+        MethodOverloadWithJustCulltureInfoAsExtraParameter(a, b, CultureInfo.CurrentCulture);
+    }
+
+    public static void MethodOverloadWithJustCulltureInfoAsExtraParameter(int a, int b, CultureInfo provider)
+    {
+        Console.WriteLine(string.Format(provider, """"));
+    }
+}",
+            GetCSharpResultAt(9, 9, SpecifyCultureInfoAnalyzer.Rule, "CultureInfoTestClass2.MethodOverloadWithJustCulltureInfoAsExtraParameter(int, int)", "CultureInfoTestClass2.Method()", "CultureInfoTestClass2.MethodOverloadWithJustCulltureInfoAsExtraParameter(int, int, System.Globalization.CultureInfo)"));
+        }
+
+        [Fact]
+        public void CA1304_NoDiagnostics_CSharp()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Globalization;
+
+public class CultureInfoTestClass2
+{
+    public static void Method()
+    {
+        // No Diag - Inherited CultureInfo
+        MethodOverloadHasInheritedCultureInfo(""Foo"");
+        // No Diag - Since the overload has more parameters apart from CultureInfo
+        MethodOverloadHasMoreThanCultureInfo(""Foo"");
+        // No Diag - Since the CultureInfo parameter is neither as the first parameter nor as the last parameter
+        MethodOverloadWithJustCulltureInfoAsInbetweenParameter("""", """");
+    }
+
+    public static void MethodOverloadHasInheritedCultureInfo(string format)
+    {
+        MethodOverloadHasInheritedCultureInfo(new DerivedCultureInfo(""""), format);
```

Comment:
Is this test testing what it claims to? Despite the name, you haven't actually derived the class DerivedCultureInfo from CultureInfo.